atlasdb-client/src/main/java/com/palantir/atlasdb/stream/AbstractGenericStreamStore.java

```diff
@@ -101,11 +100,33 @@ private InputStream tryGetStream(Transaction transaction, ID id, StreamMetadata
             loadSingleBlockToOutputStream(transaction, id, 0, ios);
             return ios.getInputStream();
         } else {
-            File file = loadToNewTempFile(transaction, id, metadata);
-            return new DeleteOnCloseFileInputStream(file);
+            return makeStream(id, metadata);
```

Comment:
Yes. The original transaction is assumed to have been closed by the time we get to loading the second block.